hw_9_corr.py

```diff
@@ -0,0 +1,287 @@
+    def quantity(self):
+        return self._quantity
+
+    @ quantity.setter
+    def quantity(self, value):
+        self._quantity = value
+
+
+class Warehouse:
+    def __init__(self):
+        self.stock = []
+
+    def find_material(self, name):
+        if self.stock:
+            for material in self.stock:
+                if material.name == name:
+                    return material
+        return None
+
+    def add_material(self, name, details, quantity, availability, price):
+        material = Material(name, details, quantity, availability, price)
+        self.stock.append(material)
+
+    def remove_material(self, material):
+        material = self.find_material(material)
+        if material:
+            self.stock.remove(material)
+
+    def current_quantity(self, material):
+        material = self.find_material(material)
+        if material:
+            print('Current stock of {0} is {1}'.format(material.name, material.quantity))
+        else:
+            print('Material does not exist.')
+
+    def current_quantity_all(self):
+        if self.stock:
+            for material in self.stock:
+                self.current_quantity(material.name)
+        else:
+            print('Warehouse is empty.')
+
+
+class Category(Warehouse):
+    def __init__(self, name):
+        self.name = name
+        self.members = []
+        super().__init__()
+
+    def add_to_category(self, material):
+        if isinstance(material, Material):
+            self.members.append(material)
+            material.category = self.name
+        else:
+            print('Material does not belong to material class.')
+
+    def print_category(self):
+        id = 0
+        if self.members:
+            for material in self.members:
+                id += 1
+                print(id, 'Material name: ', material.name, 'Material quantity :', material.quantity)
+
+
+class Basket():
+    def __init__(self):
+        self.content = {}
+        self.total = 0
+
+    def add_to_basket(self, material, value):
+        if isinstance(material, Material):
+            if material.quantity >= value:
+                self.content[material] = value
+                self.total += value * material.price
+            else:
+                raise ValueError('Material quantity in Warehouse is not enough to process your order.')
+        else:
+            raise TypeError('Material has to be an instance of class Material.')
+
+    def print_basket(self):
+        id = 0
+        if self.content:
+            print('Basket')
+            for material in self.content:
+                id += 1
+                print(id, 'Material: ', material.name, 'Quantity: ', self.content[material], 'Price: ', material.price,
+                      'Total: ', self.content[material] *  material.price)
+            print('Grand Total: ', self.total)
+
+
+class Order(Basket):
+    def __init__(self):
+        self.items = {}
+        global order_id
+        order_id += 1
```

Comment:
move to method like `_generate_order_id`

Reply:
Method _generate_order_id has been created.